src/ContinuumWorld.jl

```diff
@@ -32,70 +30,74 @@ struct CircularRegion
-const default_regions = [CircularRegion(Vec2(3.5, 2.5), 0.5),
-                         CircularRegion(Vec2(3.5, 5.5), 0.5),
-                         CircularRegion(Vec2(8.5, 2.5), 0.5),
-                         CircularRegion(Vec2(7.5, 7.5), 0.5)]
+
+const default_regions = [
+    CircularRegion(Vec2(3.5, 2.5), 0.5),
+    CircularRegion(Vec2(3.5, 5.5), 0.5),
+    CircularRegion(Vec2(8.5, 2.5), 0.5),
+    CircularRegion(Vec2(7.5, 7.5), 0.5),
+]
+
 const default_rewards = [-10.0, -5.0, 10.0, 3.0]
 
 @with_kw struct CWorld <: MDP{Vec2, Vec2}
-    xlim::Tuple{Float64, Float64}                   = (0.0, 10.0)
-    ylim::Tuple{Float64, Float64}                   = (0.0, 10.0)
-    reward_regions::Vector{CircularRegion}          = default_regions
-    rewards::Vector{Float64}                        = default_rewards
-    terminal::Vector{CircularRegion}                = default_regions
-    stdev::Float64                                  = 0.5
-    actions::Vector{Vec2}                           = cardinal
-    discount::Float64                               = 0.95
+    xlim::Tuple{Float64, Float64} = (0.0, 10.0)
+    ylim::Tuple{Float64, Float64} = (0.0, 10.0)
+    reward_regions::Vector{CircularRegion} = default_regions
+    rewards::Vector{Float64} = default_rewards
+    terminal::Vector{CircularRegion} = default_regions
+    stdev::Float64 = 0.5
+    actions::Vector{Vec2} = cardinal
+    discount::Float64 = 0.95
 end
 
 POMDPs.actions(w::CWorld) = w.actions
 POMDPs.discount(w::CWorld) = w.discount
 
 function POMDPs.transition(w::CWorld, s::AbstractVector, a::AbstractVector)
     ImplicitDistribution(w, s, a) do w, s, a, rng
-        return s + a + w.stdev*randn(rng, Vec2)
+        s + a + w.stdev * randn(rng, Vec2)
+    end
+end
+
+function POMDPs.transition(w::CWorld, s::AbstractVector, a::AbstractVector, rng::AbstractRNG)
+    ImplicitDistribution(w, s, a) do w, s, a, rng
+        s + a + w.stdev * randn(rng, Vec2)
     end
 end
 
-function POMDPs.reward(w::CWorld, s::AbstractVector, a::AbstractVector, sp::AbstractVector) # XXX inefficient
+function POMDPs.reward(w::CWorld, s::AbstractVector, a::AbstractVector, sp::AbstractVector)
     rew = 0.0
-    for (i,r) in enumerate(w.reward_regions)
+    for (i, r) in enumerate(w.reward_regions)
         if sp in r
             rew += w.rewards[i]
         end
     end
     return rew
 end
 
-function POMDPs.isterminal(w::CWorld, s::Vec2) # XXX inefficient
-    for r in w.terminal
-        if s in r
-            return true
-        end
-    end
-    return false
+function POMDPs.isterminal(w::CWorld, s::Vec2)
+    any(r -> s in r, w.terminal)
 end
 
 struct Vec2Distribution
-    xlim::Tuple{Float64,Float64}
-    ylim::Tuple{Float64,Float64}
-    d::Product #for support functions
-
-    function Vec2Distribution(xlim::Tuple{Float64,Float64}, ylim::Tuple{Float64,Float64})
+    xlim::Tuple{Float64, Float64}
+    ylim::Tuple{Float64, Float64}
+    d::Product
```

Comment:
The comment should not be removed as it may be helpful in understanding why this is included in the type.